src/main/java/it/gov/pagopa/pu/debtpositions/util/Utilities.java

```diff
@@ -58,4 +63,11 @@ public static String getRandomIUD() {
   public static String getRandomicUUID() {
     return UUID.randomUUID().toString().replace("-", "");
   }
+
+  public static String generateRandomIupd(String orgFiscalCode) {
+    ThreadLocal<SimpleDateFormat> dateFmtUUID = ThreadLocal.withInitial(() -> new SimpleDateFormat("yyMMddHHmmssSSS"));
+    return String.format("%x", Long.valueOf(orgFiscalCode)) + SEP +
+      StringUtils.remove(StringUtils.substringBeforeLast(StringUtils.substringAfter(UUID.randomUUID().toString(), SEP), SEP), SEP) +
+      SEP + String.format("%x", Long.valueOf(dateFmtUUID.get().format(new Date())));
```

Comment:
Please, don't use old java date-time sdk... Use LocalDateTime instead.
Then you can a DateTimeFormatter, which is thread-safe, thus you can avoid to use ThreadLocal and store it as a constant